src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs

```diff
@@ -39,6 +41,79 @@ namespace Microsoft.IdentityModel.Tokens
     /// </summary>
     internal class TokenUtilities
     {
+        /// <summary>
+        /// Creates a dictionary from a list of Claim's.
+        /// </summary>
+        /// <param name="claims"> A list of claims.</param>
+        /// <returns> A Dictionary representing claims.</returns>
+        internal static IDictionary<string, object> CreateDictionaryFromClaims(IEnumerable<Claim> claims)
+        {
+            var payload = new Dictionary<string, object>();
+
+            if (claims == null)
+                return payload;
+
+            foreach (Claim claim in claims)
+            {
+                if (claim == null)
+                    continue;
+
+                string jsonClaimType = claim.Type;
+                object jsonClaimValue = claim.ValueType.Equals(ClaimValueTypes.String, StringComparison.Ordinal) ? claim.Value : GetClaimValueUsingValueType(claim);
+                object existingValue;
+
+                // If there is an existing value, append to it.
+                // What to do if the 'ClaimValueType' is not the same.
+                if (payload.TryGetValue(jsonClaimType, out existingValue))
+                {
+                    IList<object> claimValues = existingValue as IList<object>;
+                    if (claimValues == null)
+                    {
+                        claimValues = new List<object>();
+                        claimValues.Add(existingValue);
+                        payload[jsonClaimType] = claimValues;
+                    }
+
+                    claimValues.Add(jsonClaimValue);
+                }
+                else
+                {
+                    payload[jsonClaimType] = jsonClaimValue;
+                }
+            }
+
+            return payload;
+        }
+
+        internal static object GetClaimValueUsingValueType(Claim claim)
+        {
+            if (claim.ValueType == ClaimValueTypes.Boolean && bool.TryParse(claim.Value, out bool boolValue))
```

Comment:
most likely, the ValueType will be ClaimValueTypes.String, perhaps we should check that first.

Reply:
I think that's a good idea if its most likely string. I made the change.